apps/showcase/src/helpers/file-system/file-system.helper.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  DirectoryNode,
+  FileNode,
+  FileSystemTree,
+  SymlinkNode,
+} from '@webcontainer/api';
+
+/** Resources to get file content */
+export type TrainingResource = {
+  /** Resource path */
+  path: string;
+  /** Resource content */
+  content: string;
+};
+
+/**
+ * Check a resource can be cast as a TrainingResource
+ * @param resource
+ */
+export const isTrainingResource = (resource: any): resource is TrainingResource => {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- needed to check object type to be able to cast it
+  return typeof resource?.content === 'string' && typeof resource?.path === 'string';
+};
+
+/**
+ * Generate a file system tree composed of the deep merge of all the resources passed in parameters
+ * @param resources Sorted list of path and content to load. If a file is defined several time, the last occurrence
+ * overrides the others
+ */
+export function getFilesContent(resources: TrainingResource[]) {
+  return (resources.reduce((fileSystemTree: FileSystemTree, resource) => {
+    const sanitizedPath = `./${resource.path.replace(new RegExp('^[.]/?'), '')}`;
```

Comment:
Could be simplify to
```suggestion
    const sanitizedPath = resource.path.replace(/^[.]\/?/, './');
```
Don't you have an issue with `path` if value starts with `..` or if `path` is a file starting with a `.` like `.editorconfig`